diff --git a/app.js b/app.js
index 8a1634a..d535c84 100644
--- a/app.js
+++ b/app.js
@@ -13,9 +13,9 @@ const cors = require("cors");
 const querystring = require("querystring");
 const cookieParser = require("cookie-parser");
 
-const client_id = "CLIENT_ID"; // Your client id
-const client_secret = "CLIENT_SECRET"; // Your secret
-const redirect_uri = "REDIRECT_URI"; // Your redirect uri
+const client_id = 'id'; // Your client id
+const client_secret = 'sec'; // Your secret
+const redirect_uri = 'http://localhost:5000/callback'; // Your redirect uri
 
 /**
  * Generates a random string containing numbers and letters
@@ -47,7 +47,7 @@ app.get("/login", function(req, res) {
   res.cookie(stateKey, state);
 
   // your application requests authorization
-  let scope = "user-read-private user-read-email";
+  let scope = 'user-read-private user-read-email user-library-read playlist-modify-private user-read-currently-playing user-modify-playback-state playlist-read-collaborative user-read-playback-state streaming playlist-read-private';
   res.redirect(
     "https://accounts.spotify.com/authorize?" +
       querystring.stringify({
@@ -66,6 +66,9 @@ app.get("/callback", function(req, res) {
   let code = req.query.code || null;
   let state = req.query.state || null;
   let storedState = req.cookies ? req.cookies[stateKey] : null;
+  let client_id = client_id || MISSING;
+  let client_secret = client_secret || MISSING;
+  let redirect_uri = redirect_uri || 'http://localhost:5000/callback';
 
   if (state === null || state !== storedState) {
     res.redirect("/#" + querystring.stringify({ error: "state_mismatch" }));
@@ -99,12 +102,25 @@ app.get("/callback", function(req, res) {
             res.redirect(
               "/#" + querystring.stringify({ access_token, refresh_token })
             );
+	  console.log('REKEY=' + refresh_token);
           })
           .catch(e => {
             res.redirect(
               "/#" + querystring.stringify({ error: e.response.data })
             );
           });
+	  function refresh() {
+	  axios({
+          	method: "post",
+          	url: "https://accounts.spotify.com/api/token",
+		client_id: client_id,
+		client_secret: client_secret,
+		grant_type: 'refresh_token',
+		refresh_token: refresh_token,
+          	headers: { "Content-Type": "application/x-www-form-urlencoded" }
+          })
+	  }
+	  setInterval(refresh, 1000 * 56 * 56);
       })
       .catch(e => console.error(e.response.data));
   }
@@ -137,6 +153,4 @@ app.get("/refresh_token", function(req, res) {
       console.error(e.response.data);
     });
 });
-
-console.log("Listening on 8888");
-app.listen(8888);
+app.listen(5000);
