diff --git a/authorization_code/app.js b/authorization_code/app.js
index 8a1634a..a01b0ef 100644
--- a/authorization_code/app.js
+++ b/authorization_code/app.js
@@ -13,9 +13,19 @@ const cors = require("cors");
 const querystring = require("querystring");
 const cookieParser = require("cookie-parser");
 
-const client_id = "CLIENT_ID"; // Your client id
-const client_secret = "CLIENT_SECRET"; // Your secret
-const redirect_uri = "REDIRECT_URI"; // Your redirect uri
+const client_id = 'id'; // Your client id
+const client_secret = 'sec'; // Your secret
+const redirect_uri = 'http://localhost:5000/callback'; // Your redirect uri
+
+// We need to pass though some variables, into '/public/index.html's script, I've tried everything, last try
+const fs = require('fs');
+const http = require('http');
+//
+
+       // Let's hope this shit works
+       var content = ('const client_id = ' + '"' + client_id + '";' + '\n' + 'const client_secret = ' + '"' + client_secret + '";' + '\n' + 'console.log("client_id is: " + client_id + "\\n client_secret is: " + client_secret);');
+       fs.writeFile('./public/consts.js', content, function (err) {
+                if (err) throw err;});
 
 /**
  * Generates a random string containing numbers and letters
@@ -47,7 +57,7 @@ app.get("/login", function(req, res) {
   res.cookie(stateKey, state);
 
   // your application requests authorization
-  let scope = "user-read-private user-read-email";
+  let scope = 'user-read-private user-read-email user-library-read playlist-modify-private user-read-currently-playing user-modify-playback-state playlist-read-collaborative user-read-playback-state streaming playlist-read-private';
   res.redirect(
     "https://accounts.spotify.com/authorize?" +
       querystring.stringify({
@@ -99,12 +109,25 @@ app.get("/callback", function(req, res) {
             res.redirect(
               "/#" + querystring.stringify({ access_token, refresh_token })
             );
+	 console.log('REKEY=' + refresh_token);
           })
           .catch(e => {
             res.redirect(
               "/#" + querystring.stringify({ error: e.response.data })
             );
           });
+         function refresh() {
+         axios({
+               method: "post",
+               url: "https://accounts.spotify.com/api/token",
+               client_id: client_id,
+               client_secret: client_secret,
+               grant_type: 'refresh_token',
+               refresh_token: refresh_token,
+               headers: { "Content-Type": "application/x-www-form-urlencoded" }
+          })
+         }
+         setInterval(refresh, 1000 * 56 * 56);
       })
       .catch(e => console.error(e.response.data));
   }
@@ -137,6 +160,4 @@ app.get("/refresh_token", function(req, res) {
       console.error(e.response.data);
     });
 });
-
-console.log("Listening on 8888");
-app.listen(8888);
+app.listen(5000);
